roofit/roofitcore/src/RooArgSet.cxx

```diff
@@ -104,8 +104,10 @@ void RooArgSet::cleanup()
 
 void* RooArgSet::operator new (size_t bytes)
 {
-  //This will fail if a derived class uses this operator
-  assert(sizeof(RooArgSet) == bytes);
+  // To make sure that derived classes don't use this operator
```

Comment:
Is not this a different behaviour than before ? 
And before was failing only in debug mode, was this intended ?

Reply:
Yes, the behavior is different now. Before, the assert would fail if the size doesn't match (i.e. because if the class derives from and extends RooArgSet), now it falls back to not using the memory pool for derived classes.

Before, not using the memory pool for `RooSetProxy` was achieved by overriding `operator new` again in `RooSetProxy`, which is not needed after this change because the `RooArgSet::operator new` will fallback itself to the original operator new for derived classes.